classes/Product.php

```diff
@@ -7329,11 +7329,25 @@ public static function getIdProductAttributeByIdAttributes($idProduct, $idAttrib
             $idAttributes = [(int) $idAttributes];
         }
 
+        $emptyAttributeGroups = [];
+        foreach ($idAttributes as $idAttributeGroup => $idAttribute) {
+            if (is_numeric($idAttributeGroup) && empty($idAttribute)) {
+                $emptyAttributeGroups[] = (int) $idAttributeGroup;
+            }
+        }
+
+        $idAttributes = array_filter($idAttributes);
+
```

Comment:
```suggestion
        $idAttributes = array_filter($idAttributes);

        $emptyAttributeGroups = [];
        foreach ($idAttributes as $idAttributeGroup => $idAttribute) {
            if (is_numeric($idAttributeGroup) && empty($idAttribute)) {
                $emptyAttributeGroups[] = (int) $idAttributeGroup;
            }
        }
```

Reply:
Hi @Progi1984, Thanks for the feedback.
Do you just change the order of the `array_filter`?
I do not remember exactly, but I think we need to know the empty groups before `array_filter($idAttributes)`.
Do you test your changes? I'm not sure about this change.

Reply:
Oh shit. I have misread the code.